### ISSUE TITLE
Str::template() Removes Token-Like Fields That Don't Have Mappings

### ISSUE THREAD

**Issue by neildaniels** [getkirby/kirby#2307]
**Describe the bug**  
When using the `Str::template()` method, if the source string has "token"-like parts to it (but are not meant to be replaced) then they will be unexpectedly removed.

```php
Str::template('From {{ b }} to {{ a }}, maybe {{ c }}.', [
    'a' => 'there',
    'b' => 'here'
]);
//   Actual: From here to there, maybe .
// Expected: From here to there, maybe {{ c }}.
```

This actual behavior differs from K2-behavior, and to me it seems like a bug.

**Kirby Version**  
`3.3.0`

**Comment by afbora** [comment]
Only one fallback can be assigned for now:

````php
Str::template('From {{ b }} to {{ a }}, maybe {{ c }}.', [
    'a' => 'there',
    'b' => 'here'
], '{{ c }}');
````

Maybe we can enhance it to support the array:

````php
Str::template('From {{ b }} to {{ a }}, maybe {{ c }}.', [
    'a' => 'there',
    'b' => 'here'
], [
    'a' => 'fallback a',
    'b' => 'fallback b',
    'c' => 'fallback c'
]);
````

**Comment by neildaniels** [comment]
Unfortunately, that doesn't really provide the previous behavior I'm looking for.

I often use `Str::template()` with the source string being pulled in from a page field.

For example: `Str::template($page->someField(), [ /* some available options */])`

And the reason I find it annoying that all non-matching `{{ var }}` patterns get removed is that it makes it difficult for an editor to determine if that value is actually empty, or they're using a replacement token that doesn't exist.

I'm not sure what most people would expect, but I think `null`/`false` for `$replacement` should be interpreted to mean "leave non-matching tokens in the string", and an empty string `''` should be interpreted as the current behavior of "removing" the non-matching tokens.

While I do like the ability to have multiple fallbacks, that's already easy to achieve with the following.
```php
$data = [
    'a' => 'there',
    'b' => 'here'
];
$fallbacks = [
    'a' => 'fallback a',
    'b' => 'fallback b',
    'c' => 'fallback c'
];
$mergedData = array_merge($fallbacks, $data);

Str::template('From {{ b }} to {{ a }}, maybe {{ c }}.', mergedData);
```

I think allowing an array in that parameter leads to too many questions like "what should `[]` be interpreted to mean?"

**Comment by lukasbestle** [comment]
I agree with @neildaniels on the "array as `$fallback`" feature.

The original issue is fixed by #2309.

**Comment by bastianallgeier** [comment]
✅
